### PR TITLE
Avoid taking snapshot during journal suspension

### PR DESCRIPTION
**Pull request by bf8086**
When journal is suspended, snapshot should not be taken because some entries included up to the snapshot index might not have been applied by the suspended journal applier.

### CODE REVIEW

**Comment by bf8086** [comment]
@gpang @LuQQiu PTAL.

File: core/server/common/src/test/java/alluxio/master/journal/raft/RaftJournalTest.java [hunk 1]
```diff
@@ -139,6 +139,50 @@ public void suspendCatchupResume() throws Exception {
...
+              .setInodeLastModificationTime(
+                  File.InodeLastModificationTimeEntry.newBuilder().setId(entryCount).build())
+              .build());
+    }
+
+    // Ask the follower to do a snapshot.
+    mFollowerJournalSystem.checkpoint();
+
+    // Restart the follower.
+    mFollowerJournalSystem.stop();
```
Comment (LuQQiu) [inline core/server/common/src/test/java/alluxio/master/journal/raft/RaftJournalTest.java:175]: why added restart follower here

**Review by LuQQiu — COMMENTED** [review]
Thanks, left a minor comment

File: core/server/common/src/test/java/alluxio/master/journal/raft/RaftJournalTest.java (hunk 1 above)
Comment (bf8086) [reply core/server/common/src/test/java/alluxio/master/journal/raft/RaftJournalTest.java:175]: Restart causes the follower snapshot to be replayed so its content can be verified.

**Review by gpang — APPROVED** [review]
LGTM

**Review by LuQQiu — APPROVED** [review]
LGTM

**Comment by bf8086** [comment]
alluxio-bot, merge this please.

**Pull request merged**